pkg/ide/vscode/vscode.go

```diff
@@ -106,7 +181,48 @@ func (o *VsCodeServer) InstallExtensions() error {
 
 			break
 		}
-	} else if o.releaseChannel == ReleaseChannelInsiders {
+
+		return binPath
+	}
+
+	if o.flavor == FlavorCursor {
+		// check legacy location `$HOME/.cursor-server/bin`
+		binDir := filepath.Join(location, "bin")
+		for {
+			entries, err := os.ReadDir(binDir)
+			if err != nil || len(entries) == 0 {
+				o.log.Infof("Read dir %s: %v", binDir, err)
+				o.log.Info("Wait until cursor-server is installed...")
+				// check new location `$HOME/.cursor-server/cli/servers/Stable-<version>/server/bin/cursor-server`
+				newBinPath, err := o.findCodeServerBinary(location)
+				if err != nil {
+					o.log.Infof("Read new location %s: %v", location, err)
+					o.log.Info("Wait until cursor-server is installed...")
+					time.Sleep(time.Second * 3)
```

Comment:
this could block forever. Maybe we should set a MAX_ATTEMPTS and loop over this? If we are waiting for cursor to be installed maybe a couple of minutes would be long enough?

Reply:
good point, added a timeout of 10 minutes for now. Should be plenty based on experience. If we ever encounter issues we could expose this setting in the context similar to the Agent inject timeout